pkgs/build-support/rust/import-cargo-lock.nix

```
@@ -89,18 +93,22 @@ let
     '';
     fetchurl {
       name = "crate-${pkg.name}-${pkg.version}.tar.gz";
-      url = "https://crates.io/api/v1/crates/${pkg.name}/${pkg.version}/download";
+      url = "${registryUrl}/v1/crates/${pkg.name}/${pkg.version}/download";
       sha256 = checksum;
     };
 
+  registries = {
+    "registry+https://github.com/rust-lang/crates.io-index" = "https://crates.io/api";
```

should we limit all the extra registries to be "registry+"? are there legit use cases where the source wouldn't start with "registry+"?

```suggestion
    "https://github.com/rust-lang/crates.io-index" = "https://crates.io/api";
```

---

As far as I know, all other use cases start with `git+` and they are already covered by other parameters to this function. Happy to chop off the `registry+` prefix in this API.